### PR TITLE
add nodeSelector and tolerations for the ClusterManager operator

### PR DESCRIPTION
**Pull request by zhiweiyin318**
this PR is to add nodeSelector and Tolerations for pods to run on the specified nodes like master, ocp infra nodes.

Signed-off-by: Zhiwei <zyin@redhat.com>

### CODE REVIEW

File: operator/v1/types.go [hunk 1]
```diff
@@ -40,6 +41,14 @@ type ClusterManagerSpec struct {
 	// +optional
 	// +kubebuilder:default=quay.io/open-cluster-management/placement
 	PlacementImagePullSpec string `json:"placementImagePullSpec,omitempty"`
+
+	// NodeSelector defines which Nodes the Pods are scheduled on. The default is an empty list.
+	// +optional
```
Comment (qiujian16) [inline operator/v1/types.go:46]: put NodeSelector and Tolerations in a struct.

File: operator/v1/types.go [hunk 2]
```diff
@@ -40,6 +41,14 @@ type ClusterManagerSpec struct {
...
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+
+	// Tolerations defines the tolerations for the Pods. The default is an empty list.
```
Comment (qiujian16) [inline operator/v1/types.go]: I think you would need to ref this to the definition of toleration. Current doc is not clear.

Comment (zhiweiyin318) [reply operator/v1/types.go]: done. put them in NodePlacement

File: operator/v1/types.go (hunk 1 above)
Comment (zhiweiyin318) [reply operator/v1/types.go:46]: done. put them in NodePlacement

File: operator/v1/types.go (hunk 2 above)
Comment (zhiweiyin318) [reply operator/v1/types.go]: done

**Pull request merged**